### PR TITLE
External IP constraints

### PR DESCRIPTION
**Pull request by tallclair**
Add constraint template & samples for restricting service external IPs to an allowlist.

### CODE REVIEW

File: library/general/externalip/template.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+              items:
+                type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8sexternalips
+
+        violation[{"msg": msg}] {
+          input.review.kind.kind == "Service"
+          input.review.kind.group == ""
```
Comment (ritazh) [inline library/general/externalip/template.yaml:25]: let's make sure the rego here is consistent with the src.rego

Comment (tallclair) [reply library/general/externalip/template.yaml:25]: oops, this was just commented out in the source for debugging. fixed.

File: library/general/externalip/template.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          allowedIPs := {ip | ip := input.parameters.allowedIPs[_]}
```
Comment (maxsmythe) [inline library/general/externalip/template.yaml:26]: Do we want to use CIDRs here? We could use `net.cidr_contains_matches`:

https://www.openpolicyagent.org/docs/latest/policy-reference/#net

Comment (tallclair) [reply library/general/externalip/template.yaml:26]: Oh neat! I think both would be useful. Is it OK if we proceed with this as-is, and I'll follow up with a PR to add an `allowedCIDRs` parameter?

Comment (maxsmythe) [reply library/general/externalip/template.yaml:26]: That works!

**Review by maxsmythe — APPROVED** [review]
LGTM

**Review by ritazh — APPROVED** [review]
lgtm

**Pull request merged**